### PR TITLE
py-triangle: New port

### PR DESCRIPTION
**Pull request by smithsp**
This pull request supersedes the Portfile submitted in https://trac.macports.org/ticket/47941

### CODE REVIEW

File: python/py-triangle/Portfile [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+github.setup        drufat triangle 20160203 v
+
+name                py-triangle
+
+categories-append   science math
+
+platforms           darwin
+supported_archs     noarch
+license             LGPL
```
Comment (neverpanic) [inline python/py-triangle/Portfile]: Please specify the version of LGPL, and if it's 'or any later version' append a `+`.

File: python/py-triangle/Portfile [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    Documentation at http://dzhelil.info/triangle/index.html
+
+distname            triangle-${version}
+checksums           md5     ef7e121d51a8dc2bd5163a0db11b2b16 \
+                    rmd160  4aa893ccaf47db733d11bb87b2bd5793e2e8a987 \
+                    sha256  419fb6d773e6c187a3b850cfd74970dfd31804a5c9281207fd4198384fb1b27a
+
+python.versions     27 34 35
+
+if {$subport != $name} {
```
Comment (neverpanic) [inline python/py-triangle/Portfile]: Since you're comparing strings, not integers, Tcl prefers `$subports ne $name`.

**Comment by smithsp** [comment]
@neverpanic 
Thanks for the review.  I put the license as I did because the author of the python package only lists it as GNU LGPL: https://github.com/drufat/triangle/search?utf8=✓&q=license

**Comment by smithsp** [comment]
@neverpanic Let me know if you want me squash and force push.

**Comment by neverpanic** [comment]
Well, the original license of triangle seems to be (from README):
```
﻿These programs may be freely redistributed under the condition that the
copyright notices (including the copy of this notice in the code comments
and the copyright notice printed when the `-h' switch is selected) are
not removed, and no compensation is received.  Private, research, and
institutional use is free.  You may distribute modified versions of this
code UNDER THE CONDITION THAT THIS CODE AND ANY MODIFICATIONS MADE TO IT
IN THE SAME FILE REMAIN UNDER COPYRIGHT OF THE ORIGINAL AUTHOR, BOTH
SOURCE AND OBJECT CODE ARE MADE FREELY AVAILABLE WITHOUT CHARGE, AND
CLEAR NOTICE IS GIVEN OF THE MODIFICATIONS.  Distribution of this code as
part of a commercial system is permissible ONLY BY DIRECT ARRANGEMENT
WITH THE AUTHOR.  (If you are not directly supplying this code to a
customer, and you are instead telling them how they can obtain it for
free, then you are not required to make any arrangement with me.)
```

In a nutshell, we don't remove the copyright notices, so we are free to redistribute. But then again, the license text goes on to explain that private, research, and institutional use is free, but commercial is not (what if you're using it commercially but keep the copyright notices? Is that covered by the first sentence, or by the latter?).

This software is a prime example of why you should not make up your own license, unless you're a lawyer (and probably not even then). Since py-triangle contains the triangle code and this license text seems incompatible with LGPL-3.0 (which does not allow additional restrictions), I suggest we use `license LGPLv3 Noncommercial Restrictive` so that MacPorts won't distribute precompiled binaries, but still allow local compilation.

As for the process, please squash and force-push.

**Comment by neverpanic** [comment]
Also, since merging this would close https://trac.macports.org/ticket/47941, please include `Closes: https://trac.macports.org/ticket/47941` in your commit message.

**Comment by smithsp** [comment]
Note that in https://github.com/drufat/triangle/issues/15, @drufat indicated to use LGPL-3 as you had surmised.

**Comment by smithsp** [comment]
Also, I squashed and pushed.

**Comment by neverpanic** [comment]
@drufat only clarified the license of py-triangle as LGPL-3, assuming the `triangle` code does not have a indicated license. I've checked the triangle source code, and that has a license (the one mentioned above).

Because the LGPL-3 code is linked together with this code (and it's not dynamic linking, but multiple object files getting linked together), the LGPL-3's viral effect would spread to the `triangle` code, which would require you to release it under the same conditions. Unfortunately, this isn't possible, because the triangle source code contains clauses that seem incompatible with LGPL-3 to me (but I'm not a lawyer), which means we must refrain from distributing binary copies of this.

Putting `license LGPL-3` in the file would make MacPorts distribute binary copies of this module, which would violate its license. That's why I asked you to also add `Noncommercial Restrictive`, which prevents our build systems from distributing binaries and somewhat accurately represents the custom license of the `triangle` source code.

**Comment by smithsp** [comment]
Thanks for taking care of this @neverpanic.  I went on vacation...

**Pull request merged**